src/Core/Application/RealTime/UseCase/FindService/FindService.php

```diff
@@ -73,6 +77,106 @@ public function __invoke(
         int $serviceId,
         FindServicePresenterInterface $presenter
     ): void {
+        $servicegroups = [];
+
+        $this->debug(
+            "Searching details for service",
+            [
+                "id" => $serviceId
+            ]
+        );
+
+        if ($this->contact->isAdmin()) {
+            $host = $this->hostRepository->findHostById($hostId);
+            if ($host === null) {
+                $this->critical(
+                    "Host not found",
+                    [
+                        'id' => $hostId,
+                        'userId' => $this->contact->getId()
+                    ]
+                );
+                $presenter->setResponseStatus(new HostNotFoundResponse());
+                return;
+            }
+            $service = $this->repository->findServiceById($serviceId);
+            if ($service === null) {
+                $this->critical(
+                    "Service not found",
+                    [
+                        'id' => $serviceId,
+                        'hostId' => $hostId,
+                        'userId' => $this->contact->getId()
+                    ]
+                );
+                $presenter->setResponseStatus(new ServiceNotFoundResponse());
+                return;
+            }
+
+            $servicegroups = $this->servicegroupRepository->findAllByHostIdAndServiceId(
+                $hostId,
+                $serviceId
+            );
```

Comment:
Not needed to handle a servicegroupNotFound if $servicegroups === null ?

Reply:
Servicegroups can be empty it is not mandatory. Not all services are linked to a servicegroup